blocks/rich-text/index.js

```diff
@@ -358,21 +352,21 @@ export default class RichText extends Component {
 		}
 	}
 
-	fireChange() {
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.state.empty ? [] : this.savedContent );
-	}
-
 	/**
 	 * Handles any case where the content of the tinyMCE instance has changed.
 	 */
 	onChange() {
-		// Note that due to efficiency, speed and low cost requirements isDirty may
-		// not reflect reality for a brief period immediately after a change.
-		if ( this.editor.isDirty() ) {
-			this.fireChange();
+		this.savedContent = this.getContent();
+		this.props.onChange( this.state.empty ? [] : this.savedContent );
```

Comment:
Minor: I made a small change in the other PR here, I assign `[]` to this.savedContent if it's empty to avoid `updateContent` to be called on the next render.